replica/database.hh

```diff
@@ -327,6 +327,40 @@ struct cf_stats {
     uint64_t total_view_updates_on_wrong_node = 0;
 };
 
+/*!
+ * \brief consumption_unit_counter represents a mechanism for tracking the consumption units.
+ *
+ * Consumption units are simplified representations of user operations within the system.
+ * Each operation performed by a user consumes a specific number of
+ * units, depending on its complexity and resource requirements.
+ *
+ * The `consumption_unit_counter` class encapsulates the concept of consumption units.
+ *
+ * This class is **front-end-protocol-agnostic**, meaning it does not depend on any specific front-end protocol.
+ * Each protocol that utilizes this class will define how the consumption units are updated and managed within
+ * its context.
+ *
+ * Consumption units serve two primary purposes:
+ *
+ * 1. Billing: They provide a standardized way to quantify resource usage, enabling accurate and transparent
+ *    billing based on the capacity units consumed by a user's operations.
+ *
+ * 2. Usage Limitations: They help enforce usage policies by limiting the number of capacity units
+ *    a user can consume within a given time frame.
+ *
+ */
+class consumption_unit_counter {
+    uint64_t _total_units = 0;
+public:
+    uint64_t get_units() const noexcept {
+        return _total_units;
+    }
+    uint64_t add_units(uint64_t units) noexcept {
+        _total_units += units;
+        return _total_units;
+    }
```

Comment:
Why add a class? Isn't uint64_t sufficient?

Reply:
The added encapsulation is to make it easy to separate enterprise and open-source code.

Reply:
This isn't a good explanation, patches should stand on their own.

Reply:
The patch stands on its own; it does exactly what it says it does. Yes, it's a bit more complicated than a simple counter, but it's 100% functional.